File: BlinkID/android/src/main/java/com/microblink/flutter/OverlaySettings.java
package com.microblink.flutter;

import androidx.annotation.NonNull;

import java.util.Map;

public class OverlaySettings {
    Boolean useFrontCamera = false;
    Boolean flipFrontCamera = false;

    public OverlaySettings(Map<String, Object> args){
        this.useFrontCamera = (Boolean) args.get("useFrontCamera");
        this.flipFrontCamera = (Boolean) args.get("flipFrontCamera");
    }

}

File: BlinkID/android/src/main/kotlin/com/microblink/blinkid/flutter/OverlaySettings.kt
package com.microblink.blinkid.flutter

class OverlaySettings {
    val useFrontCamera: Boolean
    val flipFrontCamera: Boolean

    constructor(args: Map<String, Any?>, defaultSettings: OverlaySettings) {
        useFrontCamera = args["useFrontCamera"] as Boolean? ?: defaultSettings.useFrontCamera
        flipFrontCamera = args["flipFrontCamera"] as Boolean? ?: defaultSettings.flipFrontCamera
    }

    constructor(useFrontCamera: Boolean, flipFrontCamera: Boolean) {
        this.useFrontCamera = useFrontCamera
        this.flipFrontCamera = flipFrontCamera
    }

}